### PR TITLE
VMware: add new module vmware_host_dns_facts

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE

 - New Module Pull Request
 
##### COMPONENT NAME

lib/ansible/modules/cloud/vmware/vmware_host_dns_facts.py
test/integration/targets/vmware_host_dns_facts/aliases
test/integration/targets/vmware_host_dns_facts/tasks/main.yml

##### ANSIBLE VERSION

```
2.5devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
@pdellaert @tchernomax Could you please review this ?

File: lib/ansible/modules/cloud/vmware/vmware_host_dns_facts.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+from ansible.module_utils.vmware import vmware_argument_spec, PyVmomi
+
+
+class VmwareDnsFactsManager(PyVmomi):
+    def __init__(self, module):
+        super(VmwareDnsFactsManager, self).__init__(module)
+        cluster_name = self.params.get('cluster_name', None)
+        esxi_host_name = self.params.get('esxi_hostname', None)
+        self.hosts = []
+        if cluster_name:
```
Comment (pdellaert) [inline lib/ansible/modules/cloud/vmware/vmware_host_dns_facts.py:90]: As this piece of code is being added to many modules (getting hosts), perhaps we can put that in the vmware module utils at some point? (no need now, just an improvement for the future :))

**Review by pdellaert — APPROVED** [review]
LGTM

File: lib/ansible/modules/cloud/vmware/vmware_host_dns_facts.py (hunk 1 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_host_dns_facts.py:90]: Make sense. I will add it in next PR. This thought came in my mind as well. But I was too lazy to do it. anyways. ;)

**Comment by Akasurde** [comment]
@pdellaert Thanks for review.

**Pull request merged**